### DISCUSSION TITLE
weight for custom field choices

### DISCUSSION THREAD

**Discussion by saeedhmt** [netbox-community/netbox#6750] [Q&A] [answered]
Hi.
in earlier version of netbox, when we defined a custom field of type selection or multi selection, we could add weight to the choices and choices would be sorted by the weights but now this feature is not there. is there a reason that this feature is removed?
![cf-weight](https://user-images.githubusercontent.com/58300378/125569311-d544d307-db34-4eb3-a009-0ee174b10b5c.png)

**Comment by candlerb** [comment, marked as answer]
There used to be a separate model for custom field choices.  This has now been collapsed, so a single CustomField model contains all the data about that custom field.  Within that model, a single field [`choices`](https://github.com/netbox-community/netbox/blob/develop/netbox/extras/models/customfields.py#L102-L107) contains a comma-separated list of choice values.

I believe that the choices will be displayed in the order that you put them in that list - which makes the weights redundant, as weights were purely for sorting the list.  Now you just give the list in whatever order you want.

Does your experience show that is not the case?  What Netbox version are you running?

**Reply by saeedhmt** [reply]
well suppose you have a custom field and some choices for it. after a while you add new choices and maybe it's easier to sort them by weight instead of looking for their position in the list of choices.
and the version of netbox is 2.8.7

**Reply by candlerb** [reply]
Sorry, I don't think that's a valid use case.  What you are saying is: "I have a list of options.  I want them to appear in my chosen order.  However, I can't be bothered to insert a new option into the correct place in the existing list of options".

**Reply by saeedhmt** [reply]
you're right. thanks for your time.
